cbits/is-valid-utf8.c

```diff
@@ -0,0 +1,110 @@
+#pragma GCC push_options
+#pragma GCC optimize("-O2")
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+// 0x80 in every 'lane'.
+static uint64_t const high_bits_mask = 0x8080808080808080ULL;
+
+int is_valid_utf8(uint8_t const *const src, size_t const len) {
```

Comment:
This function should really be named something `bytestring` (or at least Haskell) specific. Keep in mind that the C symbol namespace is process-global.